api/include/opentelemetry/logs/severity.h

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+#include "opentelemetry/version.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace logs
+{
+
+/**
+ * Severity Levels assigned to log events, based on Log Data Model
+ */
+enum class Severity : uint8_t
+{
+  kInvalid = 0,
+  kTrace   = 1,
```

Comment:
This looks cool and quite readable. But `If the first enumerator does not have an initializer, the associated value is zero. For any other enumerator whose definition does not have an initializer, the associated value is the value of the previous enumerator plus one.` ... basically you do not have to write `= 0`, `= 1`, etc.. Is there a coding style standard that forces us to write more? Since we can safely write less.

Reply:
That makes sense, here since we don't have any non-continuous severity number mappings and the first one starts at 0, we could simplify by removing all the initializers. Will change this in the next commit!